bin/elmstatic.js

```diff
@@ -276,12 +276,14 @@ function generateHtml(elmJs, pageOrPost) {
         return "<!doctype html>" + R.replace(/citatsmle-script/g, "script", dom.window.document.body.innerHTML)
 }
 
-// String -> HtmlPage | PostHtmlPage -> ()/Effects
+// String -> HtmlPage | PostHtmlPage -> Promise
 function writeHtmlPage(page) {
     const outputPath = R.endsWith("index", page.outputPath) ?
         page.outputPath + ".html" : Path.join(page.outputPath, "index.html")
-    Fs.mkdirsSync(Path.dirname(outputPath))
-    Fs.writeFileSync(outputPath, page.html)
+    return Fs.mkdirs(Path.dirname(outputPath))
+        .then(() => {
+            return Fs.writeFile(outputPath, page.html)
+        })
```

Comment:
I'm not keen to add the complexity of async functions unless there is a tangible performance improvement – I've deliberately stuck to sync functions so far in order to keep the code simple. My impression was that the run time is dominated by running JS. Do you have performance numbers for this?

Reply:
`time elmstatic build`
### first run
latest elmstatic:
real    0m4.088s
user    0m1.639s
sys     0m0.813s
--------
@eimfach/elmstatic
real    0m2.603s
user    0m1.359s
sys     0m0.480s
### second run
latest elmstatic
real    0m2.293s
user    0m1.381s
sys     0m0.511s
-------------
@eimfach/elmstatic
real    0m1.308s
user    0m1.112s
sys     0m0.325s

Reply:
I am using the elmstatic boilerplate, I thought you could test the performance locally on korban.net ? `npm i @eimfach/elmstatic`

Reply:
writing sync is always blocking the main execution. You also could generate the html in the first place and the write sync, but not `html -> write sync -> html -> write sync ... and so on`